src/components/RequestModal/MovieRequestModal.tsx

```diff
@@ -215,20 +223,32 @@ const MovieRequestModal: React.FC<RequestModalProps> = ({
           is4k ? messages.pending4krequest : messages.pendingrequest,
           { title: data?.title }
         )}
-        onOk={() => (showEditButton ? updateRequest() : cancelRequest())}
+        onOk={() =>
+          canManageRequests
+            ? updateRequest(true)
+            : canEditRequests
+            ? updateRequest()
+            : cancelRequest()
+        }
         okDisabled={isUpdating}
         okText={
-          showEditButton
+          canManageRequests
+            ? intl.formatMessage(messages.editAndApprove)
+            : canEditRequests
             ? intl.formatMessage(messages.edit)
             : intl.formatMessage(messages.cancel)
         }
-        okButtonType={showEditButton ? 'primary' : 'danger'}
+        okButtonType={
+          canManageRequests ? 'success' : canEditRequests ? 'primary' : 'danger'
+        }
```

Comment:
```suggestion
        okButtonType={showEditButton ? 'primary' : 'danger'}
```
I don't think it's necessary to turn the button green, it looks kinda odd IMO 🤔

Reply:
Wasn't really sure about this neither. The colour of the primary button in the modal is purple but the approve button on the request page is green.